[TASK] Documentation: Nginx reverse proxy, Virtual_* variable (#50)
pluswerk/php-dev: docs/nginx-reverse-proxy.md

# Nginx Reverse Proxy

As already mentioned, this project relies on [pluswerk/docker-global](https://github.com/pluswerk/docker-global).

Here is an information about the VIRTUAL_* variable clarified.

## VIRTUAL_HOST: Virtual host (your domain)

> So that the nginx reverse proxy knows which domain belongs to which Docker container there is the VIRTUAL_HOST variable.

Reviewer: ```suggestion
Through the VIRTUAL_HOST variable the nginx reverse proxy knows which domain belongs to which Docker container.
```

##### Variant A

> You can simply specify this domain directly without regular expressions. See variant A.

Reviewer 2: ```suggestion
You can specify this domain directly without regular expressions. See variant A.
```

As described more clearly here: https://css-tricks.com/words-avoid-educational-writing/
Words like `simply` shouldn't be present in Documentation or any code related writings.

> You can simply specify this domain directly without regular expressions. See variant A.

Reviewer: ```suggestion
You can specify this domain directly without regular expressions. Look at variant A.
```

##### Regular expression information

> However, I would recommend a bit of regular expressions.

Reviewer: ```suggestion
We do recommend the use of regular expressions.
```

The regular expression starts with a ~ and ends with a $$. So do not wonder why it's two dollars.

If you have a DNS in between, you have the option to use multiple domains.
For example, domain.vm & domain.vmd with different IP addresses.
One goes to the local host, the other into a virtual machine. The possibilities are endless.

##### Variant B

For example, if you have a TYPO3 website, you could use variant B.
With this you can use all subdomains if you have several websites in the project.

##### Variant C + D

But what if you have two Docker containers with the same domain?

For example, you have a Symfony website domain.vm, but also a phpbb forum on forum.domain.vm.
This means you create two directories with a docker-compose.yml for each project.

You use variant C for the Symfony website, because domain.vm and www.domain.vm should be used.

For the forum you use variant D, so you have forum.domain.vm.

Finally, I can say that variant C or D is the better choice in most cases.
But since every system is different, you can of course do what suits you.

File docker-compose.yml:

> ```yaml

Reviewer 2: if the example file is on top I don't have to scroll so often.

Reviewer: i think so too, the examples should go directly under the variants

Author: Done
